### DISCUSSION TITLE
Relative vs Absolute Imports - How should IDEs be configured?

### DISCUSSION THREAD

**Discussion by bchilcott** [mkosir/typescript-style-guide#4] [Q&A]
Below is the guidance on using Relative vs Absolute imports:
> - Relative imports ./sortItems must be used when importing files within the same feature, that are 'close' to each other, which also allows moving feature around the codebase without introducing changes in these imports.
> - Absolute imports @common/utils must be used in all other cases.

IDEs (as far as I'm aware, at least) don't provide this kind of granular control, which makes it difficult. WebStorm, for example, can be configured to use relative imports always, never, or only between members of an alias.

VSCode is similar, but can automatically choose the shortest path, whether that's between modules or not.

I'm aware this is a Typescript style guide and not an IDE configuration one - but do you have any recommendations on how to configure an IDE to best support these rules? A solution in WebStorm at least would be to configure an alias for each feature individually, but this wouldn't work the same in VSCode.

**Comment by mkosir** [comment]
Hey @bchilcott, thanks on this! I never really paid to much attention on how my IDE (VSCode) handles it, if it happens it gets it "wrong", I of course manually edit it.
But thats a thats a great point, as similar guide topic [Test Tooling](https://mkosir.github.io/typescript-style-guide/#test-tooling), would be awesome to include it into guide, feel free to come up with PR!
